app/models/play.rb

```diff
@@ -0,0 +1,20 @@
+# == Schema Information
+#
+# Table name: plays
+#
+#  id        :bigint           not null, primary key
+#  played_at :datetime         not null
+#  track_id  :bigint           not null
+#  user_id   :bigint           not null
+#
+class Play < ApplicationRecord
+  belongs_to :track
+  belongs_to :user
+
+  validates :played_at, presence: true
+  validate :play_cannot_be_in_future
```

Comment:
I'm not entirely sure about this validation. It seems logical in theory, but slight clock drifts can make this very annoying for users. I also don't think faking plays to be in the future is something we should be worried about?